sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobAsyncClient.java

```diff
@@ -187,22 +188,23 @@
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.storage.blob.specialized.BlockBlobAsyncClient.upload#Flux-int-int}
+     * {@codesnippet com.azure.storage.blob.specialized.BlockBlobAsyncClient.upload#Flux-ParallelTransferOptions}
      *
      * @param data The data to write to the blob. Unlike other upload methods, this method does not require that the
      * {@code Flux} be replayable. In other words, it does not have to support multiple subscribers and is not expected
      * to produce the same values across subscriptions.
-     * @param blockSize The size of each block that will be staged. This value also determines the size that each buffer
-     * used by this method will be and determines the number of requests that need to be made. The amount of memory
-     * consumed by this method may be up to blockSize * numBuffers. If block size is large, this method will make fewer
-     * network calls, but each individual call will send more data and will therefore take longer.
-     * @param numBuffers The maximum number of buffers this method should allocate. Must be at least two. Typically, the
-     * larger the number of buffers, the more parallel, and thus faster, the upload portion of this operation will be.
-     * The amount of memory consumed by this method may be up to blockSize * numBuffers.
+     * @param parallelTransferOptions {@link ParallelTransferOptions} to use to buffered upload.
+     * The block size is the size of each block that will be staged. This value also determines the size that each
+     * buffer used by this method will be and determines the number of requests that need to be made. The amount of
+     * memory consumed by this method may be up to blockSize * numBuffers. If block size is large, this method will make
+     * fewer network calls, but each individual call will send more data and will therefore take longer.
+     * The number of parallel transfers is the maximum number of buffers this method should allocate. Must be at least
+     * two. Typically, the larger the number of buffers, the more parallel, and thus faster, the upload portion of this
+     * operation will be. The amount of memory consumed by this method may be up to blockSize * numBuffers.
```

Comment:
I tend to try to put long descriptions like this in the javadocs for the type itself and just link to the type in the parameter description. Then you don't have to copy/paste the descriptions every time you use the parameter.